### PR TITLE
Serialize/Deserialize as String instead of json object

### PR DESCRIPTION
**Pull request by ellisjoe**
@markelliot @pnepywoda

### CODE REVIEW

**Comment by pnepywoda** [comment]
oh wow, it's so simple with just annotations. Cool! +1

File: gradle.properties [hunk 1]
```diff
@@ -5,4 +5,4 @@ jacocoVersion=0.7.1.201405082137
 
 # test compile dependencies
 junitVersion=4.11
-jacksonVersion=2.5.1
+jacksonVersion=2.4.4
```
Comment (pnepywoda) [inline gradle.properties:8]: Note that this was my suggestion. It follows https://github.com/palantir/http-remoting/pull/57 (spark needing jackson 2.4)

File: build.gradle [hunk 2]
```diff
@@ -23,8 +23,9 @@ repositories {
 }
 
 dependencies {
+    compile "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"
```
Comment (markelliot) [inline build.gradle]: only need annotations for compile, not databind.

File: src/main/java/com/palantir/ri/ResourceIdentifier.java [hunk 3]
```diff
@@ -222,6 +225,7 @@ public static boolean isValidLocator(String locator) {
      * <a href="http://docs.oracle.com/javaee/7/api/javax/ws/rs/PathParam.html">
      * query and path parameters</a>
      */
+    @JsonCreator
```
Comment (markelliot) [inline src/main/java/com/palantir/ri/ResourceIdentifier.java:223]: if we're going to include this we can mark all of service, instance, type and locator as final.

Comment (ellisjoe) [inline src/main/java/com/palantir/ri/ResourceIdentifier.java:223]: Done. and removed default constructor

File: build.gradle (hunk 2 above)
Comment (ellisjoe) [inline build.gradle]: Done

**Comment by markelliot** [comment]
worth adding a test that running through an objectmapper renders the string you want? (we know the inverse works already)

**Comment by ellisjoe** [comment]
not sure I understand. how is that different from the tests I added?

**Comment by markelliot** [comment]
your test calls #toString() which already did the right thing/testing the annotations are processed correctly requires that the ObjectMapper actually get invoked?

**Comment by markelliot** [comment]
nvm, I suck at reading

**Comment by markelliot** [comment]
Ok, this is good. I think we've basically converged on something that we probably ought to convert to immutables, but we can save that for another time.

**Comment by ellisjoe** [comment]
Yeah, was thinking the same thing.

**Pull request merged**